server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -182,6 +181,13 @@ public DocTableInfo(RelationName ident,
                         @Nullable Version versionUpgraded,
                         boolean closed,
                         Set<Operation> supportedOperations) {
+        this.notNullColumns = references.values().stream()
+            .filter(r -> !r.column().isSystemColumn())
+            .filter(r -> !primaryKeys.contains(r.column()))
+            .filter(r -> !r.isNullable())
```

Comment:
I think we can remove  `!primaryKeys.contains` check since setting `nullable` flag in Reference accounts for that

Reply:
It would cause `notNullColumns` to include primary keys too, which would change the behavior slightly.

We already had a bit of an inconsistency here. `pk1 int not null primary key` was present in `notNullColumns` but `pk1 int primary key` was not. And both had `nullable=false`